### PR TITLE
docs: glossary, keyword matching and  tool-tips

### LINKED ISSUE #33117: docs: setup glossary
This bounty is for both setting up the glossary infrastructure and populating it with most openpilot terms.

Requirements:
* a master definition file
* "glossary page" under the concepts section
* nice to have: click or hover on words defined in the glossary to get the definition on any docs.comma.ai page. bounty value doubles if this is done

wiki has a good starting point on definitions: https://github.com/commaai/openpilot/wiki/General-Terms

### PR DESCRIPTION
**Pull request by LouisVelez**
Closes #33117 

Replaced by #34231

### CODE REVIEW

**Comment by LouisVelez** [comment]
Creating a draft since there are a few issues I need to fix. And I want to review the whole doc.

**Comment by LouisVelez** [comment]
Did a few changes to allow links within the tooltip description to work. Also added a link to go to the glossary from the tooltip. 

## Glossary:
image

## Keyword matching:
image

## Tooltip: 
image

### allows highlighting/copy
image

### links in descriptions
image


Tapping out for the day.

**Comment by LouisVelez** [comment]
Just noticed the panda link is breaking when the main doc has a markdown link. Will look into it...
image

File: .github/workflows/docs.yaml [hunk 1]
```diff
@@ -26,6 +26,10 @@ jobs:
       with:
         submodules: true
 
+    - uses: actions/setup-python@v5.3.0
+      with:
+        python-version: '3.12'
+
```
Comment (maxime-desroches) [inline .github/workflows/docs.yaml:32]: You can just set the runner to ubuntu-24.04 to get python 3.12 pre-installed

Comment (LouisVelez) [reply .github/workflows/docs.yaml]: Ha! I assumed latest was the latest. Thank you, change applied. I also rebased to clean the history.

**Comment by LouisVelez** [comment]
@maxime-desroches is it ok if I add some flags for the linter to ignore the problem code? Or I'm open to other suggestions.

**Comment by maxime-desroches** [comment]
No we have to fix the code. ruff offers nice descriptions for each errors:
https://docs.astral.sh/ruff/rules/function-uses-loop-variable/
https://docs.astral.sh/ruff/rules/multi-line-implicit-string-concatenation/

**Comment by LouisVelez** [comment]
> No we have to fix the code. ruff offers nice descriptions for each errors: https://docs.astral.sh/ruff/rules/function-uses-loop-variable/ https://docs.astral.sh/ruff/rules/multi-line-implicit-string-concatenation/

That was very helpful, thank you. Code fixed, [latest static analysis run](https://github.com/commaai/openpilot/actions/runs/12190686341/job/34008270425?pr=34155#step:4:20)

File: .github/workflows/docs.yaml [hunk 2]
```diff
@@ -18,7 +18,7 @@ concurrency:
 jobs:
   docs:
     name: build docs
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
```
Comment (adeebshihadeh) [inline .github/workflows/docs.yaml:21]: revert this

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -14,23 +14,24 @@ Development is coordinated through [Discord](https://discord.comma.ai) and GitHu
 ## What contributions are we looking for?
 
 **openpilot's priorities are [safety](SAFETY.md), stability, quality, and features, in that order.**
-openpilot is part of comma's mission to *solve self-driving cars while delivering shippable intermediaries*, and all development is towards that goal. 
+openpilot is part of comma's mission to *solve self-driving cars while delivering shippable intermediaries*, and all development is towards that goal.
```
Comment (adeebshihadeh) [inline docs/CONTRIBUTING.md:17]: this should be its own PR

File: docs/car-porting/brand-port.md [hunk 4]
```diff
@@ -2,4 +2,4 @@
 
 A brand port is a port of openpilot to a substantially new car brand or platform within a brand.
 
-Here's an example of one: https://github.com/commaai/openpilot/pull/23331.
+Here's an example of one: [PR-23331](https://github.com/commaai/openpilot/pull/23331).
```
Comment (adeebshihadeh) [inline docs/car-porting/brand-port.md:5]: same, this PR shouldn't change any content

File: docs/hooks/glossary.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
+import re
+import tomllib
+
+with open("docs/glossary.toml", "rb") as f:
```
Comment (adeebshihadeh) [inline docs/hooks/glossary.py:4]: don't do this at import time

File: mkdocs.yml [hunk 6]
```diff
@@ -31,7 +36,7 @@ nav:
   - Contributing:
     - Roadmap: contributing/roadmap.md
     #- Architecture: contributing/architecture.md
-    - Contributing Guide →: https://github.com/commaai/openpilot/blob/master/docs/CONTRIBUTING.md
+    - Contributing Guide: CONTRIBUTING.md
```
Comment (adeebshihadeh) [inline mkdocs.yml:39]: split this out

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]
Nice work! TODOs
* split out anything that's not related to the glossary infra
* can you also split out the new terms added to the glossary?

**Comment by LouisVelez** [comment]
> Nice work! TODOs
> 
> * split out anything that's not related to the glossary infra
> 
> * can you also split out the new terms added to the glossary?

Just to make sure I understand correctly: 

1 PR for infra only 

1 PR for the terms additions. 

1 PR for the changes in the content file (markdowns).

Does this sound correct?

**Comment by adeebshihadeh** [comment]
Yeah, and potentially even split out the content changes per-page.

**Comment by LouisVelez** [comment]
@adeebshihadeh , going to close this PR and submit the individuals just to start clean.
